base64ct/src/lib.rs

```diff
@@ -43,13 +43,228 @@ use alloc::{string::String, vec::Vec};
 /// Padding character
 const PAD: u8 = b'=';
 
+/// Standard Base64 encoding.
+///
+/// ```text
+/// [A-Z]      [a-z]      [0-9]      +     /
+/// 0x41-0x5a, 0x61-0x7a, 0x30-0x39, 0x2b, 0x2f
+/// ```
+pub mod base64 {
```

Comment:
Since you introduce modules, I think it could be worth to introduce a `b64` module and move all `unpadded` functions to it while keeping the padded methods in `base64`. It will allow us to remove the `padded`/`unpadded` suffixes, thus improving the naming a bit.

Reply:
The problem is there are four total combinations this PR supports:

1. Padded standard Base64
2. Unpadded standard Base64
3. Padded URL-safe Base64
4. Unpadded URL-safe Base64

While (2) is "B64", (4) is also an important practical encoding as well: it's used by JWK.

I don't think it would make sense to refer to 4 as "URL-safe B64", as the "B64" name comes from the PHC string format which doesn't define anything for the URL-safe variant.

Another option using modules is to have a structure like:

```rust
mod base64 {
    mod padded {
        ...
    }

    mod unpadded {
        ...
    }
}

mod base64url {
    mod padded {
        ...
    }

    mod unpadded {
        ...
    }
}
```

Reply:
How about: `baes64ct::padded`, `baes64ct::unpadded`, `baes64ct::url::padded`, and `baes64ct::url::unpadded`? An alternative to the latter two could be `baes64ct::padded::url` and `baes64ct::unpadded::url`.

Reply:
Sure, that works

Reply:
Refactored the code with this module structure